bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ValueComparisonUtils.java

```diff
@@ -143,7 +173,8 @@ private static int compareFloatValues(double lhsValue, double rhsValue) {
      * @param direction Sort direction ascending/descending/""
      * @return 1 if the left hand side value > right hand side value,
      * -1 if left hand side value < right hand side value,
-     * 0 left hand side value = right hand side value
+     * 0 left hand side value = right hand side value,
+     * -999 if the left hand side value is unordered with respect to right hand side value
```

Comment:
Instead of an invalid value, shall we throw an exception, catch it and return false?